graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/views/searchtypes/pivot/buckets/OSValuesHandler.java

```diff
@@ -101,6 +105,30 @@ private Script scriptForPivots(Collection<String> pivots) {
         return new Script(scriptSource);
     }
 
+    private TermsAggregationBuilder applyOrdering(Pivot pivot, TermsAggregationBuilder terms, List<BucketOrder> ordering, OSGeneratedQueryContext queryContext) {
+        return sortsOnNumericPivotField(pivot, queryContext)
+                .map(pivotSort -> terms
+                        .subAggregation(AggregationBuilders.max(SORT_HELPER).field(pivotSort.field()))
```

Comment:
I guess the `max` annotation is some kind of a trick that we need for numeric sorting, because scripted terms return string values for numeric fields? If it is so, I'd at least leave some comment about that, so that after some months nobody asks why do we have a max aggregation here.

Reply:
Good idea. Just for the explanation: _Any_ metric aggregation will work here, it can be avg/min/max. We just need an aggregation that returns a numeric value that we can sort on.